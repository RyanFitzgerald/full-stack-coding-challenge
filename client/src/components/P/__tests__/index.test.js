import React from 'react';

import P from '../index';

describe('<P />', () => {
  it('should match the snapshot', () => {
    const renderedComponent = renderer.create(
      <P>Test</P>
    ).toJSON();
    
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render the correct element', () => {
    const renderedComponent = shallow(
      <P/>
    );
    
    expect(renderedComponent.type()).toEqual('p');
  });

  it('should render its text', () => {
    const children = 'Text';
    const renderedComponent = shallow(
      <P>{children}</P>
    );
    
    expect(renderedComponent.contains(children)).toBe(true);
  });
});