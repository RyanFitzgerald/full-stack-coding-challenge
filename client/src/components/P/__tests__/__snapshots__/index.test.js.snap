// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<P /> should match the snapshot 1`] = `
.c0 {
  margin: 0;
}

<p
  className="c0"
>
  Test
</p>
`;
